src/core/manager.c

```diff
@@ -5302,6 +5302,16 @@ int manager_allocate_idle_pipe(Manager *m) {
         return 1;
 }
 
+bool manager_was_dependency_cycle(const Manager *m) {
+        const Unit *u;
+
+        HASHMAP_FOREACH(u, m->units)
+                if (u->was_on_dependency_cycle)
+                        return true;
+
+        return false;
```

Comment:
hmm, O(n) for something that mostly is a per-manager state? seems to be a bit much for systems with gazillions of units. after all this will be called by GetAllProperties() bus call, which random client stuff just calls...

maybe just maintain a counter in the Manager object, that counts the number of such units